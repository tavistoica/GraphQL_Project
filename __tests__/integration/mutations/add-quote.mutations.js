const { createTestClient } = require("apollo-server-testing");
const { gql } = require("apollo-server");
const {
  database: { url },
} = require("../../../src/core/config/test.config");
const { Server } = require("../../../src/bootstrap/server.bootstrap");
const { Database } = require("../../../src/bootstrap/database.bootstrap");

const server = new Server();
const database = new Database();

function doMutation() {
  return gql`
    mutation AddQuotes {
      addQuote(input: { phrase: "smenarie", quoter: "mafiotu" }) {
        id
        phrase
      }
    }
  `;
}

function getQuery() {
  return gql`
    query getQuotes {
      quotes {
        quoter
        phrase
      }
    }
  `;
}

async function createContext() {
  return {
    doQuotesMutation: doMutation(),
    getQuotesQuery: getQuery(),
  };
}

beforeEach(async () => {
  await database.bootstrap({ url });
  server.bootstrap();
});
afterEach(() => database.reset());

it("addQuote Mutation use case Test", async () => {
  const { doQuotesMutation, getQuotesQuery } = await createContext();
  const { query, mutate } = createTestClient(server.apolloClient);
  await mutate({
    mutation: doQuotesMutation,
  });
  const res = await query({
    query: getQuotesQuery,
  });
  expect(res).toMatchSnapshot();
});
