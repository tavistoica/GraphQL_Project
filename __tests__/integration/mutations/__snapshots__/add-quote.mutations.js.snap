// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addQuote Mutation use case Test 1`] = `
Object {
  "data": Object {
    "quotes": Array [
      Object {
        "phrase": "smenarie",
        "quoter": "mafiotu",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
